src/app/(proper_react)/redesign/(authenticated)/user/dashboard/fix/security-recommendations/SecurityRecommendationsLayout.tsx

```diff
@@ -58,12 +63,64 @@ export function SecurityRecommendationsLayout(
   // The non-null assertion here should be safe since we already did this check
   // in `./[type]/page.tsx`:
   const { title, illustration, content, exposedData } = pageData!;
+  const hasExposedData = exposedData.length > 0;
+
+  // TODO: Write unit tests MNTOR-2560
+  /* c8 ignore start */
+  const handlePrimaryButtonPress = async () => {
+    const securityRecommendatioBreachClasses: Record<
+      SecurityRecommendationTypes,
+      | (typeof SecurityRecommendationDataTypes)[keyof typeof SecurityRecommendationDataTypes]
+      | null
+    > = {
+      email: SecurityRecommendationDataTypes.Email,
+      phone: SecurityRecommendationDataTypes.Phone,
+      ip: SecurityRecommendationDataTypes.IP,
+      done: null,
+    };
+
+    const dataType = securityRecommendatioBreachClasses[props.type];
+    // Only attempt to resolve the breaches if the following conditions are true:
+    // - There is a matching data class type in this step
+    // - The current step has unresolved exposed data
+    // - There is no pending breach resolution request
+    if (!dataType || !hasExposedData || isResolving) {
+      return;
+    }
+
+    setIsResolving(true);
+    try {
+      const response = await fetch("/api/v1/user/breaches/bulk-resolve", {
+        method: "PUT",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ dataType }),
+      });
+
+      const result = await response.json();
+      if (!result?.success) {
+        throw new Error(
+          `Could not resolve breach data class of type: ${props.type}`,
+        );
+      }
+
+      const isCurrentStepSection = Object.values(stepMap).includes(nextStep.id);
+      const nextRoute = isCurrentStepSection
+        ? nextStep.href
+        : "/redesign/user/dashboard/fix/security-recommendations/done";
+      router.push(nextRoute);
+    } catch (_error) {
+      setIsResolving(false);
```

Comment:
Same Sentry Q for this `catch` too.

Reply:
[Same as above.](https://github.com/mozilla/blurts-server/pull/3791#discussion_r1404570173)